Add search functionality to MembersOverview

Fabi911 opened the pull request.

Fabi911 committed:
Add search functionality to MembersOverview

Introduced a search input that filters members based on multiple fields, including name, email, and address. Modified the DataGrid to display only the filtered results.

esgoet commented on frontend/src/Pages/MembersOverview.tsx:
```diff
@@ -76,7 +95,7 @@ export default function MembersOverview() {
 
             {membersDB &&
-                <SyledDataGrid rows={membersDB} columns={columns} getRowId={(row) => row.memberId} initialState={{
+                <SyledDataGrid rows={searchMembers()} columns={columns} getRowId={(row) => row.memberId} initialState={{
```
```suggestion
                <StyledDataGrid rows={searchMembers()} columns={columns} getRowId={(row) => row.memberId} initialState={{
```

esgoet commented on frontend/src/Pages/MembersOverview.tsx:
```diff
@@ -15,6 +15,8 @@ import AccountBoxIcon from '@mui/icons-material/AccountBox';
+    console.log(searchMembers());
 
     function fetchMembers(): void {
         axios.get('api/members')
```
usually requires / before api, but if you haven't run into any issues, then obv doesn't matter
```suggestion
        axios.get('/api/members')
```

esgoet commented on frontend/src/Pages/MembersOverview.tsx:
```diff
@@ -60,13 +62,30 @@ export default function MembersOverview() {
+
+    function searchMembers(): Member[] {
+        return membersDB?.filter(member => {
+            return member.name.toLowerCase().includes(search.toLowerCase()) ||
```
this is such a miniscule operation so I really doubt it even matters in terms of performance, but I believe it's better practice to just make a variable for lowerCaseSearch here and then use that, rather than calling .toLowerCase() each time?
```suggestion
            const lowerCaseSearch = search.toLowerCase();
            return member.name.toLowerCase().includes(lowerCaseSearch) ||
```

esgoet reviewed (approved):
lgtm, just a few optional comments

Fabi911 committed:
Add search field and layout improvements to MembersOverview

Introduced a search field to enhance member filtering functionality. Refactored layout for better code readability and user interface improvements. Adjusted fetch URL and corrected styling details.
